### PR TITLE
com.endlessm.HatchPreviewer

### PR DESCRIPTION
**Pull request by ptomato**
Flatpak package of https://github.com/endlessm/hatch-previewer

Development tool used in building apps with packaged offline content; see http://endlessm.github.io/eos-knowledge-lib/docs/master/

### CODE REVIEW

File: com.endlessm.HatchPreviewer.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    "branch": "stable",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "hatch-previewer",
+    "separate-locales": false,
+    "finish-args": [
+        "--device=dri",
+        "--filesystem=home:ro",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland"
```
Comment (TingPing) [inline com.endlessm.HatchPreviewer.json]: Doesn't support Wayland.

**Comment by TingPing** [comment]
bot, build com.endlessm.HatchPreviewer

**Comment by TingPing** [comment]
Upstream changes for `com.endlessm.HatchPreviewer.appdata.xml`:

Missing a `  <launchable type="desktop-id">com.endlessm.HatchPreviewer..desktop</launchable>` line.

```
<icon type="remote" height="256" width="256">https://raw.githubusercontent.com/endlessm/hatch-previewer/master/icons/256x256/hatch-previewer.png</icon>
<icon type="remote" height="128" width="128">https://raw.githubusercontent.com/endlessm/hatch-previewer/master/icons/128x128/hatch-previewer.png</icon>
<icon type="remote" height="64" width="64">https://raw.githubusercontent.com/endlessm/hatch-previewer/master/icons/64x64/hatch-previewer.png</icon>
```

These can be removed, appstream finds the icons since they are installed.

**Comment by ptomato** [comment]
Amended, thanks for the quick review.

**Comment by nedrichards** [comment]
bot, build com.endlessm.HatchPreviewer

File: com.endlessm.HatchPreviewer.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                /* This file is autogenerated by: flatpak-npm-generator.py package-lock.json */
+                "generated-sources.json",
+                {
+                    "type": "script",
+                    "dest-filename": "hatch-previewer",
+                    "commands": [ "npm start --prefix=/app/main $@" ]
+                }
+            ],
+            "build-commands": [
+                "npm install --prefix=main --offline --cache=/run/build/hatch-previewer/npm-cache/",
```
Comment (TingPing) [inline com.endlessm.HatchPreviewer.json]: Being ignorant of `npm`, can you not directly do `--prefix=/app/main`?

**Comment by nedrichards** [comment]
https://flathub.org/builds/#/builders/1/builds/2291

**Comment by ptomato** [comment]
If there was a build log there it looks like it's been wiped...

**Comment by nedrichards** [comment]
bot, build com.endlessm.HatchPreviewer

**Comment by nedrichards** [comment]
building here https://flathub.org/builds/#/builders/1/builds/2369

**Comment by ramcq** [comment]
Rebuilding again at https://flathub.org/builds/#/builders/1/builds/2425

**Comment by ramcq** [comment]
arch64 failed on network fail, https://flathub.org/builds/#/builders/70/builds/21 is the retry

**Comment by nedrichards** [comment]
bot, build com.endlessm.HatchPreviewer

**Comment by ptomato** [comment]
Here's a perhaps-too-wide list of people that could be admins of this app: @dnohales @ebassi @manuq @ptomato @sgnn7 @tchx84 @xjuan

**Pull request closed without merging**